### PR TITLE
Several unit tests for subscription streaming

### PR DESCRIPTION
**Pull request by antban**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/zalando/nakadi/pull/280" target="_blank">ReviewNinja

File: src/main/java/de/zalando/aruha/nakadi/service/subscription/ExactWeightRebalancer.java [hunk 1]
```diff
@@ -70,6 +70,12 @@
     }
 
     static int[] splitByWeight(final int itemCount, final int[] weigths) {
+        if (itemCount < weigths.length) {
```
Comment (rcillo) [inline src/main/java/de/zalando/aruha/nakadi/service/subscription/ExactWeightRebalancer.java]: What if I there are 2 partitions and 3 sessions? Isn't it a valid result to have `[1, 1, 0]` ?

File: src/main/java/de/zalando/aruha/nakadi/service/subscription/StreamingContext.java [hunk 2]
```diff
@@ -78,61 +87,52 @@ public long getKafkaPollTimeout() {
...
 
+    void streamInternal(final State firstState) throws InterruptedException {
         // Add first task - switch to starting state.
-        switchState(new StartingState());
+        switchState(firstState);
 
-        while (null != currentState) {
-            final Runnable task = taskQueue.poll(1, TimeUnit.MINUTES);
+        while (currentState != DEAD_STATE) {
+            // Wait forever
```
Comment (rcillo) [inline src/main/java/de/zalando/aruha/nakadi/service/subscription/StreamingContext.java]: 1 hour is not forever.

File: src/main/java/de/zalando/aruha/nakadi/service/subscription/ExactWeightRebalancer.java (hunk 1 above)
Comment (antban) [inline src/main/java/de/zalando/aruha/nakadi/service/subscription/ExactWeightRebalancer.java]: No, it's not valid.

File: src/main/java/de/zalando/aruha/nakadi/service/subscription/model/Partition.java [hunk 3]
```diff
@@ -59,19 +62,29 @@ public Partition toState(final State state, final String session, final String n
...
+     * @param existingSessionIds
+     * @return
+     */
     public Partition moveToSessionId(final String sessionId, final Collection<String> existingSessionIds) {
         switch (state) {
             case UNASSIGNED:
                 return toState(State.ASSIGNED, sessionId, null);
             case ASSIGNED:
-                if (sessionId.equals(this.session) || !existingSessionIds.contains(this.session)) {
+                if (sessionId.equals(this.session)) { // ??
```
Comment (rcillo) [inline src/main/java/de/zalando/aruha/nakadi/service/subscription/model/Partition.java]: `??` ?

File: src/main/java/de/zalando/aruha/nakadi/service/subscription/StreamingContext.java (hunk 2 above)
Comment (antban) [inline src/main/java/de/zalando/aruha/nakadi/service/subscription/StreamingContext.java]: I don't think that it's a big problem to wake up once in hour.
Actually there is no 'real forever' here possible, 1 hour is relatively big compared to commit timeout, batch_timeout and stream_timeout

File: src/main/java/de/zalando/aruha/nakadi/service/subscription/model/Partition.java (hunk 3 above)
Comment (antban) [inline src/main/java/de/zalando/aruha/nakadi/service/subscription/model/Partition.java]: sorry, changing to correct version.

File: src/test/java/de/zalando/aruha/nakadi/service/subscription/state/StreamingStateTest.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        Mockito.when(contextMock.getZkClient()).thenReturn(zkMock);
+
+        final StreamParameters spMock = StreamParameters.of(
+                1000,
+                100L,
+                100,
+                100L,
+                100,
+                100,
+                100
```
Comment (v-stepanov) [inline src/test/java/de/zalando/aruha/nakadi/service/subscription/state/StreamingStateTest.java]: The builder could really help to understand which parameter is what.

**Comment by v-stepanov** [comment]
I will close this PR as all HILA stuff (including stuff from this branch) was already merged to nakadi-jvm.

**Pull request closed without merging**